### PR TITLE
Add initial version of revised "What is polywrap" section

### PR DESCRIPTION
**Pull request by pileks**
Added a FAQ section that details all questions we have come up, together with 2 diagrams showing how wraps are loaded and executed.

Diagrams made in Canva, could probably use some work, but are usable as-is.

Canva link: https://www.canva.com/design/DAFyDryD4Ms/ovUxMgAHg9sgsyEmQXBOaw/edit?utm_content=DAFyDryD4Ms&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton

### CODE REVIEW

File: src/docs/faq.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+id: faq
+title: FAQ
+---
+
+## What is Polywrap?
+
+Polywrap is a framework for porable and composable SDKs. Polywrap calls these SDKs “wraps”. Wraps are essentially WebAssembly-based modules.
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply src/docs/faq.md:8]: I added some spice! And I provided a deeper explanation of "portable", "composable", and the value we provide.

File: src/docs/faq.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- Wraps can call other wraps:
+  - Wrap metadata is imported at built-time.
+  - Wrap code is executed at run-time.
+- Wraps help secure your app:
+  - They execute in a secure WebAssembly environment.
+  - They don't have direct access to their host environment.
+
+## How do I access host capabilities?
+
+Wraps can import host functions defined by the application developer. These native functions allow wraps to access host capabilities. Polywrap packages these capabilities as “plugins”.
```
Comment (krisbitney) [inline src/docs/faq.md:54]:
```suggestion
Polywrap has a plugin system that enables application developers to allow access to host capabilities. Wraps can call host functions defined in plugins. Plugins are also infused with Polywrap superpowers--bindings that allow them to appear to the Polywrap client and be invoked just like a wrap.

Unlike wraps, plugins are not portable. Plugins are written in a Polywrap Client's native language, and must be registered with the client at runtime.
```

Comment (krisbitney) [reply src/docs/faq.md:54]: I added more to this one as well. If you like what I'm doing, let me know and I can look at the other questions more closely as well.

**Review by krisbitney — APPROVED** [review]

File: src/docs/faq.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dOrgJelli) [reply src/docs/faq.md:8]: I really like how this is written. While I think it's a bit detailed for the first part of the Q&A, I'd rather add it than not have it. We can always shift it around in the future.

**Review by dOrgJelli — APPROVED** [review]

**Pull request merged**